sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/RequestResponseChannel.java

```diff
@@ -396,13 +396,18 @@ protected Message decodeDelivery(Delivery delivery) {
     }
 
     private void settleMessage(Message message) {
-        final String id = String.valueOf(message.getCorrelationId());
-        final UnsignedLong correlationId = UnsignedLong.valueOf(id);
+        UnsignedLong correlationId;
+        if (message.getCorrelationId() instanceof UnsignedLong) {
```

Comment:
Nice!. While the Message::correlationId is _Object_, the actual type in case of Az Messaging Services is always _UnsignedLong_. So this is a nice improvement saving us from unnecessary conversion to String and then back to long in that common cases :+1: